code/game/objects/items/stacks/medical.dm

```diff
@@ -71,44 +60,51 @@
 	var/heal_flags = NONE
 
 
-/obj/item/stack/medical/heal_pack/attack(mob/living/carbon/M as mob, mob/user as mob)
+/obj/item/stack/medical/heal_pack/attack(mob/living/M, mob/living/user)
 	. = ..()
-	if(. == TRUE)
+	if(!.) // note this true/false is inverted because we want to get the limb
 		return
 
 	var/datum/limb/affecting = .
 	var/mob/living/carbon/human/patient = M //If we've got to this point, the parent proc already checked they're human
-	if(affecting.surgery_open_stage) //Checks if mob is lying down on table for surgery
-		if(patient.can_be_operated_on())
-			do_surgery(patient, user, src)
-		else
-			to_chat(user, span_notice("\The [affecting.display_name] is cut open, you'll need more than a bandage!"))
-		return
-
-	var/unskilled_penalty = (user.skills.getRating("medical") < skill_level_needed) ? 0.5 : 1
-	var/affected = heal_limb(affecting, unskilled_penalty)
 
-	generate_treatment_messages(user, patient, affecting, affected)
-	if(!affected)
+	if(affecting.limb_status & LIMB_DESTROYED)
```

Comment:
In principle the reason I never bothered checking if a limb was destroyed is because a destroyed limb is actually healthy, and so a kit should never be applicable. By the looks of things I forgot about disinfect.